shell/middleware/authenticated.js

```diff
@@ -239,26 +84,17 @@ export default async function({
     // This only needs to happen when beforeEach hook hasn't run (the initial load)
     localCheckResource = true;
 
-    store.app.router.beforeEach((to, from, next) => {
+    store.app.router.beforeEach(async(to, from, next) => {
       // NOTE - This beforeEach runs AFTER this middleware. So anything in this middleware that requires it must set it manually
-      setProduct(store, to, redirect);
+      store.dispatch('setProduct', to);
 
       next();
     });
 
     // Call it for the initial pageload
-    const redirected = setProduct(store, route, redirect);
-
-    if (redirected) {
-      return redirected();
+    if (await store.dispatch('setProduct', route)) {
```

Comment:
We don't return a redirect method anymore but we still need to exit from this middleware method

Reply:
In the other returns it's used `false`, should we keep the same?

Reply:
I'm not sure what you're asking here. This should still be the same behavior given that setProduct doesn't return a redirect method.